resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala

```diff
@@ -53,18 +53,20 @@ private[spark] class BasicDriverFeatureStep(conf: KubernetesDriverConf)
 
   // Memory settings
   private val driverMemoryMiB = conf.get(DRIVER_MEMORY)
+  private val memoryOverheadFactor = conf.get(MEMORY_OVERHEAD_FACTOR)
+    .getOrElse(conf.get(DRIVER_MEMORY_OVERHEAD_FACTOR))
```

Comment:
Use `DRIVER_MEMORY_OVERHEAD_FACTOR` and fallback to `MEMORY_OVERHEAD_FACTOR` when missing.
We probably need to use `get(key, defaultValue)` method here though.